src/libponyc/codegen/genheader.c

```diff
@@ -70,7 +70,7 @@ static int print_pointer_type(compile_t* c, printbuf_t* buf, ast_t* type)
   ast_t* typeargs = ast_childidx(type, 2);
   ast_t* elem = ast_child(typeargs);
 
-  if(is_pointer(elem) || is_maybe(elem))
+  if (is_pointer(elem) || is_nullable_pointer(elem))
```

Comment:
the space shouldn't be added here after the `if` as it violates the c formatting that is used elsewhere. (and isnt documented at the moment, sorry).